src/CFamily/CMake/EnvironmentVarsProvider.cs

```diff
@@ -91,7 +91,11 @@ public EnvironmentVarsProvider(IVsDevCmdEnvironmentProvider vsDevCmdProvider, IL
                 }
 
                 var newSettings = await vsDevCmdProvider.GetAsync(vsDevCmdScriptParams);
-                cachedVsDevCmdSettings[vsDevCmdScriptParams] = newSettings;
+
+                if (newSettings != null)
```

Comment:
Let's discuss this fix tomorrow.
We could also potentially increase the timeout of the process, but I think this would be a better approach.